pycheribuild/config/compilation_targets.py

```diff
@@ -199,9 +200,11 @@ def riscv_arch_string(self):
         # Use the insane RISC-V arch string to enable CHERI
         if self.is_baremetal():
             # Baremetal/FreeRTOS only supports softfloat
-            arch_string = "rv64imac"
+            arch_string = "rv64imac" if self.target.cpu_architecture == CPUArchitecture.RISCV64 else \
+                          "rv32imac"
         else:
-            arch_string = "rv64imafdc"
+            arch_string = "rv64imafdc" if self.target.cpu_architecture == CPUArchitecture.RISCV64 else \
+                          "rv32imacf"
```

Comment:
This is not in the right order. But why is it using F? Our P1s have no FPU.

Reply:
That was rather intended to be generic (in case we do have simulators or targets that support F in the future) if soft-float ABI isn't used. But right now, FreeRTOS and all baremetal projects only use the `riscv_softfloat_abi` and thus `rv32imacf` is never used yet

Reply:
`rv32imafc` (NB: *not* `rv32imacf`) is a strange beast. Even for RV32 the Unix variant is intended to be `rv32imafdc` with `ilp32d` as the ABI.

Reply:
Right, I changed that to `rv32imafdc`